ffi/ffi.cpp

```diff
@@ -129,35 +223,54 @@ extern "C" uint64_t sort_hash(lean_obj_arg s)
   return std::hash<Sort>()(*sort_unbox(s));
 }
 
+extern "C" uint8_t sort_isFunction(lean_obj_arg s)
+{
+  return bool_box(sort_unbox(s)->isFunction());
+}
+
 extern "C" lean_obj_res sort_getFunctionDomainSorts(lean_obj_arg s)
 {
+  CVC5_LEAN_API_TRY_CATCH_EXCEPT_BEGIN()
   std::vector<Sort> domains = sort_unbox(s)->getFunctionDomainSorts();
   lean_object* ds = lean_mk_empty_array();
   for (const Sort& domain : domains)
   {
     ds = lean_array_push(ds, sort_box(new Sort(domain)));
   }
-  return ds;
+  return except_ok(lean_box(0), ds);
+  CVC5_LEAN_API_TRY_CATCH_EXCEPT_END()
 }
 
 extern "C" lean_obj_res sort_getFunctionCodomainSort(lean_obj_arg s)
 {
-  return sort_box(new Sort(sort_unbox(s)->getFunctionCodomainSort()));
+  CVC5_LEAN_API_TRY_CATCH_EXCEPT_BEGIN()
+  return except_ok(lean_box(0),
+    sort_box(new Sort(sort_unbox(s)->getFunctionCodomainSort()))
+  );
+  CVC5_LEAN_API_TRY_CATCH_EXCEPT_END()
 }
 
 extern "C" lean_obj_res sort_getSymbol(lean_obj_arg s)
 {
-  return lean_mk_string(sort_unbox(s)->getSymbol().c_str());
+  CVC5_LEAN_API_TRY_CATCH_EXCEPT_BEGIN()

```

Comment:
cvc5 convention is to add semicolon at the end `CVC5_LEAN_API_TRY_CATCH_EXCEPT_BEGIN;`

Reply:
- [x] remove parens from all macros that take no argument
- [x] have `;` ending all macro invokations, including the one macro that does take arguments